### ISSUE TITLE
feature relations feature to feature

### ISSUE THREAD

**Issue by urapadmin** [urapadmin/kiosk#789]
I can't find where I have read this. But there were complaints that users wanted to create feature - feature relations similar to locus - locus relations. 
I think it already got explained that the relations in features are different. Those are relations between recording records and not between archaeological entities. A category mistake (for all the philosophers out there)!  
I propose to rename "relations" on the feature form to something like "part of" or some such. The headline should indicate that this works on a different categorical level.

**Comment by urapadmin** [comment]
ja, "part of" will just lead to new confusion. Term needed!

**Comment by lbestock** [comment]
This is a philosophical point you have explained to me in depth multiple times that I still always have trouble wrapping my head around. Frustrating. 

What we mean in this case is "found within survey unit X". But, ah, that is not a workable label and for the moment nothing comes to mind.

**Comment by luizaogs** [comment]
From the discussion during the meeting it sounds like this is really about locations of features across different survey units -- is "location" too abstract? Hm.

**Comment by SamKimball-BrownU** [comment]
I think this is a case for when we just need an arch scenario to test it

**Comment by urapadmin** [comment]
I think this is completely disconnected from any arch scenario. 
The relations are not relations between archaeological entities you can observe. They are rather like a unit - locus relation than like a locus-locus relation. This is a organizational relationship and not a relationship an archaeologist observes. 
It is like the relationship between the milk you buy in the store and the item "milk" on your shopping list.

**Comment by SamKimball-BrownU** [comment]
Certainly a change of terminology is in order for the current "relations". Maybe just have a "survey units" tab or some such in the feature unit. 

But, the other question, is do we need locus-style relations between features? and the answer we came to in the meeting was... not really, because of the nature of features. They tend to be more separate/distinct than, say, strata of soil. Then there is also the issue of there not really being inverse relationships.

So, the current idea, if I am correct, is to try to tag the features in order to group them (for example, I had a couple manhole covers next to each other, but distinct, in survey unit 04. I would then tag all of those features as 04-01 or some such, so that they are all grouped by tag). They are related by their location, but not necessarily a part of the same assemblage. So, it's questionable how tightly we want to classify them together. 

IF, and this is a big if, we want to do something like this, we would have potentially a UI to link units together into categories. Perhaps this would be useful in other ways? I could easily conceive wanting to link survey and excavation units together in some way, or units in a room a certain way. HOWEVER! This seems to mostly be solved by tagging those units. There is a bit more leg work on the part of the user, but it is much more flexible. It's certainly a trade off.

So, in summary, I think robust tagging is our first thing to try. It's a system that we have not used this season, so it would be nice to see it in action before we go creating new features in the software, which is what I meant by my comment.

**Comment by urapadmin** [comment]
This ticket was meant to be about the current feature relations between survey units and features and we need a term - otherwise people will have wrong expectation about what the relations to other units in feature recording. The other topic should be discussed (and I think is in discussion) somewhere else. 
"survey units" tab does not work because features can be connected to survey units and excavation(/feature recording units.

**Comment by urapadmin** [comment]
@luizaogs if with "location" you mean from where in the recording system this feature is linked, you don't make the category mistake but I have a hunch that "location" when it comes to "features" might lead others to make the same mistake again and think of the location of the actual feature in relation to other features, no? Same difference then,

**Comment by urapadmin** [comment]
linked to?

**Comment by urapadmin** [comment]
recorded in?

**Comment by luizaogs** [comment]
With "location" I was thinking of location of feature with regards to survey unit, going off of Laurel's comment above, but that might be too obscure. I personally like "recorded in" better than "linked to" -- I think that seems closer to the original intent and "linked to" is maybe too close to "relations"?

**Comment by urapadmin** [comment]
I like this ticket. Ha, take that ontology guys!

**Comment by urapadmin** [comment]
stretchy poetic reference? Wallace Stevenson's "The Snowman" (which I still would give as an answer to the silly question (nobody asks me) what my favorite poem is.

**Comment by luizaogs** [comment]
:) Not stretchy. One must have a mind of winter indeed. I like it.

**Comment by lbestock** [comment]
Oh. Oh. I thought I was going to attend to some tickets and then finish making the figures for an almost overdue article, much more easily made thanks to the file repository. But reading Wallace Stevens again makes me shiver. I shall be human and put paint on canvas instead. I like this ticket, too.

**Comment by urapadmin** [comment]
everybody likes this ticket but we still do not have a term. I will try "recorded in". Currently I am not even sure why we have that list of units within the feature unit. In almost all cases there will be exactly one unit in there. Either the survey unit or the feature recording unit. But I remember that there were reasons to allow for a feature being linked to more than one survey unit. hm.

**Comment by luizaogs** [comment]
Laurel explained this during the meeting -- if there's a long wall that goes across multiple units, for instance, one would want to be able to indicate that it was "recorded in" all of them. Or if one creates a feature outside of a unit but then needs to assign it to a new survey unit created later. I think.

**Comment by urapadmin** [comment]
Ja, I thought that was the reason, I just have a hunch it is highly theoretical. I rather see the danger that something gets recorded twice if two survey's record the same structure and then one would indeed rather need relations between the features (like equals or what have you).

Anyway, "observed in"?

**Comment by urapadmin** [comment]
"references"?

**Comment by luizaogs** [comment]
I still prefer the more physically grounded terms -- so "recorded in" or "observed in." Mostly because "references" also seems a bit too abstract, in my mind, and the point of the tab might not be clear (so then we go back to the "relations" issue). But since Laurel is the one who actually uses it she might think differently!

**Comment by luizaogs** [comment]
(Either way, seems like a good candidate for something that should be explained in the users manual?)

**Comment by lbestock** [comment]
It isn't theoretical and your suspicion is confirmed: the Semna wall was made two different features when run into in S004 and S009. Would it have been a single feature if the linking had been possible then? Would I want it to? No, I would not - those sections of the wall are 2km distant from one another and it's helpful to have the separate sections recorded separately. But then I do want a more robust way of saying they are the same thing, which currently is just in the description where they are called "part of the Semna wall". A tag would be a more flexible way than a relation and I would prefer that, I think. More usual would be something like a terrace wall found in two adjacent survey units. 

For term I agree with Luiza. Let's try "recorded in".

I'm leery of using the users manual to try to teach archaeology itself, and this strikes me as on the border.

**Comment by luizaogs** [comment]
I just meant that the function and use of the tab doesn't seem to be very intuitive. But the new name should clear up a bit of the confusion, so maybe unnecessary.

**Comment by lbestock** [comment]
I get really lost in these tickets now and am not sure where this belongs and am not in an ideal work environment currently, but I have just remembered that there certainly IS a real case where a single tag would include both excavation loci and a feature: Wheeler rooms. The loci that we excavate and Wheeler's Room as made a feature must be tagged together, that was the whole point. Let's think about what I want to see in the tag report in that particular case. Lutz, can you update URAP if needed so I can try this? Luiza, you who are good at such things, can you find the ticket where this comment actually belongs and copy the relevant part there?

**Comment by luizaogs** [comment]
Uuh well I didn't find one specifically related to Wheeler and so am not totally sure where you want this to go but these are all seemingly related and asking for similar things (do you guys condense tickets ever? Seems like that would be useful in this case): #669, #664, #377.

**Comment by urapadmin** [comment]
we have to stop letting those tickets get out of hand like this or we don't know anymore what we are talking about. For the records: we need a different way of channeling discussions and working on specifications. The wiki needs to be involved here. But I don't want to start that discussion here, either, so I try to tie some things together:

> ... I have just remembered that there certainly IS a real case where a single tag would include both excavation loci and a feature: Wheeler rooms. The loci that we excavate and Wheeler's Room as made a feature must be tagged together, that was the whole point. 

That is the answer to my question why we have "excavating/feature recording" at all. Scary, but I really could not remember. We invented it to allow you to put legacy excavation data in. And so far that is the mere purpose of this type of unit.
So your comment touches particularly #674 but also #664. @lbestock: You might want to reopen #664 because it is there that we decided that it isn't necessary to see the feature in the list of tagged loci.

**Comment by urapadmin** [comment]
To come back to the little issue here:
### 9.4.7 
- [ ] calls the tab "recorded in" 
- [ ] and has a help key (we have neglected these little helpers of late): the new config key help/help_feature_recorded_in_list controls whether a green help button appears right of the list showing the value of that help key. The help key that's in there right now is rather dumb, but I leave help texts to more native speakers.
- [ ] if the help key is empty, the green help button should disappear

**Comment by luizaogs** [comment]
Closing this one because it's all working fine, but @lbestock see above for the other tickets!
